core/kt-osrd-sim-infra/src/main/kotlin/fr/sncf/osrd/sim_infra/impl/RawInfraBuilder.kt

```diff
@@ -273,6 +274,14 @@ class RawInfraBuilderImpl : RawInfraBuilder {
         nextZones[detector] = zone
     }
 
+    override fun setNextNodes() {
+        for (trackNode in trackNodePool) {
+            for (port in trackNodePool[trackNode].ports) {
+                nextNode.getOrPut(trackNodePool[trackNode].ports[port]) { trackNode }
+            }
+        }
+    }
+
```

Comment:
Can be part of `resolveReferences`

If not, the name is weird, is sounds like a setter